src/servers/exporter.ts

```diff
@@ -108,19 +110,14 @@ const main = async () => {
     const userCount = await User.countDocuments()
     userCount_g.set(userCount)
 
-    for (const role of roles) {
-      try {
-        const wallet = await getWalletFromRole({ role, logger })
-        const balanceSats = await Wallets.getBalanceForWallet({
-          walletId: wallet.user.walletId as WalletId,
-          logger,
-        })
-        if (balanceSats instanceof Error) throw balanceSats
+    for (const index in roles) {
+      const role = roles[index]
+      const account = await account_roles[index]
```

Comment:
Could this loose coupling be tricky to maintain in the future vs. something like an object with `role` as key and `account_role` as value?

Reply:
the idea I have is to stop manually calling "role" outside of the ledger as I have been using it for now, but instead of load it at startup and keep in memory here. the loading happens here: [src/services/mongodb/index.ts](https://github.com/GaloyMoney/galoy/blob/main/src/services/mongodb/index.ts)

Reply:
this will improve efficiency because we were fetching previously the _id associated to account for every transaction